tracing-attributes/src/lib.rs

```diff
@@ -330,9 +354,41 @@ fn gen_body(
         ..
     } = sig;
 
-    let err = args.err;
     let warnings = args.warnings();
 
+    let body = gen_block(
+        block,
+        params,
+        asyncness.is_some(),
+        args,
+        instrumented_function_name,
+        variables_rebindings,
+        self_type,
+    );
+
+    quote!(
+        #(#attrs) *
+        #vis #constness #unsafety #asyncness #abi fn #ident<#gen_params>(#params) #return_type
+        #where_clause
+        {
+            #warnings
+            #body
+        }
+    )
+}
+
+// Instrument a block
+fn gen_block<'a>(
+    block: &'a Block,
+    params: &'a Punctuated<FnArg, Token![,]>,
```

Comment:
nit: surprised the explicit lifetime is necessary? does
```suggestion
fn gen_block(
    block: &Block,
    params: &Punctuated<FnArg, Token![,]>,
```
not compile here?